### ISSUE TITLE
Bug: Type conversion failure for `events` stream of `source-klaviyo` when run in PyAirbyte

### ISSUE THREAD

**Issue by aaronsteers** [airbytehq/PyAirbyte#57]
Reported [in slack](https://airbytehq-team.slack.com/archives/C06FZ238P8W/p1708526473508759).

`Could not convert 'false' with type str: tried to convert to boolean`

I checked the [`events` schema](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-klaviyo/source_klaviyo/schemas/events.json) for Klaviyo and it doesn't seem to have any boolean-typed fields, which is odd.

Here's the full traceback:

```
>>> cache = ab.get_default_cache()
>>> source.select_streams(["events"])
>>> result = source.read(cache=cache)
Started `source-klaviyo` read operation at 12:56:42...
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/gsb-eng/prog/envs/andromeda/lib/python3.11/site-packages/airbyte/source.py", line 567, in read
    cache.process_airbyte_messages(
  File "/Users/gsb-eng/prog/envs/andromeda/lib/python3.11/site-packages/airbyte/_processors.py", line 218, in process_airbyte_messages
    record_batch = pa.Table.from_pylist(stream_batch)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "pyarrow/table.pxi", line 1876, in pyarrow.lib._Tabular.from_pylist
  File "pyarrow/table.pxi", line 5327, in pyarrow.lib._from_pylist
  File "pyarrow/table.pxi", line 3969, in pyarrow.lib.Table.from_arrays
  File "pyarrow/table.pxi", line 1448, in pyarrow.lib._sanitize_arrays
  File "pyarrow/table.pxi", line 1429, in pyarrow.lib._schema_from_arrays
  File "pyarrow/array.pxi", line 344, in pyarrow.lib.array
  File "pyarrow/array.pxi", line 42, in pyarrow.lib._sequence_to_array
  File "pyarrow/error.pxi", line 154, in pyarrow.lib.pyarrow_internal_check_status
  File "pyarrow/error.pxi", line 91, in pyarrow.lib.check_status
pyarrow.lib.ArrowInvalid: Could not convert 'false' with type str: tried to convert to boolean
```

**Comment by aaronsteers** [comment]
Starting a PR here which hopefully will let us efficiently repro, and would be generalizable to any source for which we (Airbyte) have integration test creds:

- https://github.com/airbytehq/PyAirbyte/pull/58

**Comment by aaronsteers** [comment]
Resolved by #60 

Please re-open if any issues persist.
